driver_4_lights.ino

```diff
@@ -1,225 +1,345 @@
-          }
-          Serial.println("State = topleft");
-          lightState = bottomleft;
-        } else if (lightState == bottomleft) {
-          led_set(3, OFF);
-          led_set(4, OFF);
-          led_set(5, ON);
-          led_set(6, OFF);
-          if (clickState == ON) {
-            cameraClick();
-          }
-          Serial.print("State == bottomleft");
-          lightState = bottomright;
-        } else if (lightState == bottomright) {
-          led_set(3, OFF);
-          led_set(4, OFF);
-          led_set(5, OFF);
-          led_set(6, ON);
-          if (clickState == ON) {
-            cameraClick();
-          }
-          Serial.print("State == bottomright");
-          lightState = all;
-        } else {
-          lightState = all;
-        }
+unsigned long handleIndicatorLED(int state, unsigned long last_indicator_change) {
+  if (state == 0) { //Do nothing
+    if (millis() - last_indicator_change > 1000) { //1x/sec
+      Serial.println("sec");
+      if (HIGH == digitalRead(indicatorLed)) {
+        digitalWrite(indicatorLed, LOW);
+      } else {
+        digitalWrite(indicatorLed, HIGH);
+      }
+      last_indicator_change = millis();
+    }
+  } else if (state == 1) { //StepLights state
+    if (millis() - last_indicator_change > 300) { //3x/sec
+      if (HIGH == digitalRead(indicatorLed)) {
+        digitalWrite(indicatorLed, LOW);
+      } else {
+        digitalWrite(indicatorLed, HIGH);
+      }
+      last_indicator_change = millis();
+    }
+  } else if (state == 2) { //ManualStepThrough state
+    if (millis() - last_indicator_change > 100) { //10x/sec
+      if(HIGH == digitalRead(indicatorLed)) {
+        digitalWrite(indicatorLed, LOW);
+      } else {
+        digitalWrite(indicatorLed, HIGH);
       }
+      last_indicator_change = millis();
     }
-    //  digitalWrite(3, HIGH);
-    //  delay(1000);
-    //  cameraClick();
+  } else if (state == 3) { //AutoStepThrough keep on
+      if(LOW == digitalRead(indicatorLed)) {
+        digitalWrite(indicatorLed, HIGH);  //keeps the indicator LED on
+      }
+      last_indicator_change = millis();
   }
+  return last_indicator_change;
+}
 
 
-  // set the LED:
-  digitalWrite(ledPin, ledState);
-
-  // save the reading.  Next time through the loop,
-  // it'll be the lastButtonState:
-  lastButtonPinState = reading;
-
+void resetSystemState() {
+  Serial.println("I'm resetting variables and hardware to starting configuration");
+  doNoLights(); // Turn all lights off
+  lightOnMode = 0;
 }
 
-void cameraClick() {
-  Serial.println ("take my picture");
-  digitalWrite(1, HIGH);
-  digitalWrite(1, LOW);
+void takePicture() {   //shutter clicks and takes a picture
+  digitalWrite(cameraShutter, LOW);
+  delay(shutterLag);
+  Serial.println("take my picture");
+  digitalWrite(cameraShutter, HIGH);
+  delay(shutterDelay);
+  digitalWrite(cameraShutter, LOW);
+  delay(shutterLag);
 }
 
 void led_set(uint8_t number, uint8_t setting) {
```

Comment:
If you are interested in setting all of the LEDs in one shot you can set the bits on PORTB.  Not needed for the speeds you are switching at, but worth knowing about. =) https://www.arduino.cc/en/Reference/PortManipulation